client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -148,10 +148,11 @@ export const FlowRenderer: React.FC< { flow: Flow; steps: readonly StepperStep[]
 		// and are redirected to the user step.
 		const postAuthStepSlug = stepData?.nextStep ?? '';
 		if ( step.slug === PRIVATE_STEPS.USER.slug && postAuthStepSlug ) {
+			const flowSlug = flow.variantSlug ?? flow.name;
```

Comment:
Can that cause issues if the variant slug matches an existing flow name? I don't think we have checks in place to prevent that from happening.

Reply:
We don't have checks in place for this, but we'll have other (much larger!) issues if we have a collision between a flow and a variant, as only one of them will work.

More broadly, I think it would be good to deprecate the variant functionality, as it will likely keep generating unexpected bugs like this. I know we're planning to consolidate the `site-migration` and `hosted-site-migration` flows into a combined flow in the next few weeks, so that will be one less implementation. I believe most of the other flows depending on flow variants are related to domain flows.

Reply:
- Agreed on removing variants. In theory, if we write good reusable steps, and abstract any relevant flow logic, there wouldn't be a need for variants.
- Along the same line, we should also stop extending flow objects (see [this comment](https://github.com/Automattic/wp-calypso/pull/100042#issuecomment-2672177739))
- we should refactor all steps and and all flows to be oindex by their slug, which would inherently make it impossible to have two flows with the same slug, or two steps with the same slugs